gevent_async/executor.py

```diff
@@ -0,0 +1,41 @@
+import functools
+from concurrent.futures import Future
+from typing import Callable, TypeVar
+
+from gevent import threadpool
+from typing_extensions import ParamSpec
+
+T = TypeVar("T")
+P = ParamSpec("P")
+
+
+class GeventExecutor(threadpool.ThreadPoolExecutor):
+    def submit(
+        self, fn: Callable[P, T], *args: P.args, **kwargs: P.kwargs
+    ) -> Future[T]:
+        # Gevent's returned futures do not map well to Python futures, so we
+        # must translate. We can't just use set_result/set_exception because
+        # done callbacks are not always called in gevent's case and it doesn't
+        # seem to support cancel, so we instead wrap the caller function.
```

Comment:
[non-blocking/optional] I think it might be clearer to explicitly state "Gevent's returned futures do not implement all of the things expected by `concurrent.futures.Future`" rather than saying they "do not map well". I wasn't sure what the latter meant, so I wound up reading gevent code to figure out what this meant.